revolutionpi/revolution_pi.go

```diff
@@ -145,13 +149,40 @@ func (b *revolutionPiBoard) DoCommand(ctx context.Context,
 ) (map[string]interface{}, error) {
 	resp := make(map[string]interface{})
 
-	_, ok := req["getStatus"]
-	if ok {
-		err := b.controlChip.showDeviceList()
+	pinMessage, exists := req[readAddressKey]
+	if exists {
+		pinName, ok := pinMessage.(string)
+		if !ok {
+			return nil, fmt.Errorf("error performing readAddress: expected string got %v", pinMessage)
+		}
+		pin := SPIVariable{strVarName: char32(pinName)}
+		err := b.controlChip.mapNameToAddress(&pin)
 		if err != nil {
 			return nil, err
 		}
-		resp["getStatus"] = "status ok"
+		b.controlChip.logger.Debugf("pin: %#v", pin)
+		switch pin.i16uLength {
+		case 1:
+			value, err := b.controlChip.getBitValue(int64(pin.i16uAddress), pin.i8uBit)
+			if err != nil {
+				return nil, err
+			}
+			resp[pinName] = value
+		default:
```

Comment:
I don't really understand what's happening in the default case. Consider putting in a comment on that.

Reply:
added a comment in both cases